pyiceberg/table/__init__.py

```diff
@@ -545,7 +545,7 @@ def new_snapshot_id(self) -> int:
 
     def current_snapshot(self) -> Optional[Snapshot]:
         """Get the current snapshot for this table, or None if there is no current snapshot."""
-        if snapshot_id := self.metadata.current_snapshot_id:
+        if (snapshot_id := self.metadata.current_snapshot_id) is not None or isinstance(snapshot_id, int):
             return self.snapshot_by_id(snapshot_id)
```

Comment:
I think the following is easier to read, WDYT?
```suggestion
        if self.metadata.current_snapshot_id is not None:
            return self.snapshot_by_id(self.metadata.current_snapshot_id)
```

Reply:
yes, makes sense, I did not want to change the walrus operator, but this look more readable.